### PR TITLE
add milliseconds into timestamps

### LINKED ISSUE #721: change datetime formatting to include milliseconds
I propose `'%Y-%m-%dT%H:%M:%SZ'` in `utcformat` becomes `'%Y-%m-%dT%H:%M:%S.%fZ'`, note the `%f` to give more exact times.

This is because we have numerous jobs with take around a second and are queued for around a second and without `started_at` and `ended_at` being saved with ms it's impossible to get a good idea of how long their really taking.

I can see from the comments in `utcparse` that something similar was previously the case so I wanted to canvas opinions on whether this was a possibility before submitting a PR?

### PR DESCRIPTION
**Pull request by samuelcolvin**
fix #721

### CODE REVIEW

**Comment by selwin** [comment]
@samuelcolvin what are your thoughts about using unix timestamp instead of the current human readable format? It might be slightly more efficient.

**Comment by samuelcolvin** [comment]
I thought about it and indeed the async successor to rq: [arq](https://github.com/samuelcolvin/arq) uses unix timestamps, but there are major disadvantages to unix time stamps and only really one big advantage.

The **big advantage** is that you can store/parse/transfer an int which should be more efficient than a string, but only if the encoding has native support for int packing. rq uses json which encodes ints as strings, thus storing unix time stamps isn't significantly more efficient (either in terms of performance or space) than the current timestamps. arq uses msgpack to encode data in redis which can take advantage of efficient int packing so storing an int is much more efficient than a datetime string.

The **big disadvantages** of unix time stamps are:
* they're bloody hard to make any sense of as humans. For example I give you `1505836763`, make any sense to you? me neither
* There's massive ambiguity in whether unix timestamps are in seconds or ms, the two approaches are used roughly 50-50. This application would require ms, but that's even hard to read, even relatively for a human, for example: could a user realistically have filled in their profile between `1505836763321` and `1505837369987`? not so easy to read.
* in this situation the best approach might be to use seconds since 1970 as a float, but that creates a new set of problems:
  * it's not a standard format at all
  * you have to be careful when formatting floats that you don't get 8 decimal places and end up with massive strings

In summary, I've agonised about unix timestamps vs formatted datetimes in the past a lot, hence this rant. In this situation I don't think it's a good idea.
